scanners/reactnative/plain.go

```diff
@@ -1,139 +1,143 @@
+
+		schemeOption := models.NewOption(ios.SchemeInputTitle, ios.SchemeInputSummary, ios.SchemeInputEnvKey, models.TypeSelector)
+		projectPathOption.AddOption(project.RelPath, schemeOption)
+
+		for _, scheme := range project.Schemes {
+			exportMethodOption := models.NewOption(ios.DistributionMethodInputTitle, ios.DistributionMethodInputSummary, ios.DistributionMethodEnvKey, models.TypeSelector)
+			schemeOption.AddOption(scheme.Name, exportMethodOption)
+
+			for _, exportMethod := range ios.IosExportMethods {
+				iosConfig := ios.NewConfigDescriptor(project.IsPodWorkspace, project.CarthageCommand, scheme.HasXCTests, scheme.HasAppClip, exportMethod, scheme.Missing)
+				descriptor := configDescriptor{
+					hasIOS:          true,
+					hasAndroid:      hasAndroid,
+					hasTest:         hasTests,
+					hasYarnLockFile: hasYarnLockFile,
+					ios:             iosConfig,
+				}
+				descriptors = append(descriptors, descriptor)
 
-	// ios options
-	var iosOptions *models.OptionNode
-	iosDir := filepath.Join(projectDir, "ios")
-	if exist, err := pathutil.IsDirExists(iosDir); err != nil {
-		return models.OptionNode{}, warnings, err
-	} else if exist {
-		scanner.iosScanner.SuppressPodFileParseError = true
-		if detected, err := scanner.iosScanner.DetectPlatform(scanner.searchDir); err != nil {
-			return models.OptionNode{}, warnings, err
-		} else if detected {
-			options, warns, _, err := scanner.iosScanner.Options()
-			warnings = append(warnings, warns...)
-			if err != nil {
-				return models.OptionNode{}, warnings, err
+				exportMethodOption.AddConfig(exportMethod, models.NewConfigOption(descriptor.configName(), nil))
 			}
-
-			iosOptions = &options
 		}
 	}
 
-	if androidOptions == nil && iosOptions == nil {
-		return models.OptionNode{}, warnings, errors.New("no ios nor android project detected")
-	}
-	// ---
-
-	if androidOptions != nil {
-		if iosOptions == nil {
-			// we only found an android project
-			// we need to update the config names
-			lastChilds := androidOptions.LastChilds()
-			for _, child := range lastChilds {
-				for _, child := range child.ChildOptionMap {
-					if child.Config == "" {
-						return models.OptionNode{}, warnings, fmt.Errorf("no config for option: %s", child.String())
-					}
-
-					configName := configName(true, false, scanner.hasTest)
-					child.Config = configName
-				}
-			}
-		} else {
-			// we have both ios and android projects
-			// we need to remove the android option's config names,
-			// since ios options will hold them
-			androidOptions.RemoveConfigs()
-		}
+	return projectPathOption, warnings, descriptors
+}
 
+// options implements ScannerInterface.Options function for plain React Native projects.
+func (scanner *Scanner) options(project project) (models.OptionNode, models.Warnings) {
+	const defaultVariant = "Debug"
+	var (
+		rootOption     models.OptionNode
+		allDescriptors []configDescriptor
+		warnings       = models.Warnings{}
```

Comment:
NOTE: could be `warnings       models.Warnings`

Reply:
Updated.